From 030b8d4617eefa021b380d66c7c656d495c813d6 Mon Sep 17 00:00:00 2001
From: Peter Varga <pvarga@inf.u-szeged.hu>
Date: Tue, 2 Jun 2020 08:19:28 +0200
Subject: [PATCH 02/17] qtwebenginec: Fix VS2017 builds

Fixes errors like:
  error C2139: 'v8::internal::Cell': an undefined class is not
  allowed as an argument to compiler intrinsic type trait
  '__is_convertible_to'

Task-number: QTBUG-84451
Change-Id: I6299a292c41c29ad76ecdd4aaec09f28397f756f
Reviewed-by: Allan Sandfeld Jensen <allan.jensen@qt.io>
---
 chromium/v8/src/codegen/code-stub-assembler.h | 33 +++++++++++++++++++
 1 file changed, 33 insertions(+)

diff --git a/qtwebengine/src/3rdparty/chromium/v8/src/codegen/code-stub-assembler.h b/qtwebengine/src/3rdparty/chromium/v8/src/codegen/code-stub-assembler.h
index 2c838f37382..a4ed8a12f0e 100644
--- a/qtwebengine/src/3rdparty/chromium/v8/src/codegen/code-stub-assembler.h
+++ b/qtwebengine/src/3rdparty/chromium/v8/src/codegen/code-stub-assembler.h
@@ -12,13 +12,46 @@
 #include "src/common/globals.h"
 #include "src/common/message-template.h"
 #include "src/compiler/code-assembler.h"
+#include "src/objects/api-callbacks.h"
 #include "src/objects/arguments.h"
 #include "src/objects/bigint.h"
+#include "src/objects/cell.h"
+#include "src/objects/descriptor-array.h"
+#include "src/objects/embedder-data-array.h"
+#include "src/objects/feedback-cell.h"
+#include "src/objects/feedback-vector.h"
+#include "src/objects/free-space.h"
+#include "src/objects/js-break-iterator.h"
+#include "src/objects/js-collator.h"
+#include "src/objects/js-date-time-format.h"
+#include "src/objects/js-display-names.h"
+#include "src/objects/js-generator.h"
+#include "src/objects/js-list-format.h"
+#include "src/objects/js-locale.h"
+#include "src/objects/js-number-format.h"
+#include "src/objects/js-promise.h"
+#include "src/objects/js-plural-rules.h"
+#include "src/objects/js-regexp-string-iterator.h"
+#include "src/objects/js-relative-time-format.h"
+#include "src/objects/js-segment-iterator.h"
+#include "src/objects/js-segmenter.h"
+#include "src/objects/js-weak-refs.h"
+#include "src/objects/literal-objects.h"
 #include "src/objects/objects.h"
 #include "src/objects/promise.h"
+#include "src/objects/property-cell.h"
+#include "src/objects/prototype-info.h"
+#include "src/objects/regexp-match-info.h"
+#include "src/objects/scope-info.h"
 #include "src/objects/shared-function-info.h"
 #include "src/objects/smi.h"
+#include "src/objects/source-text-module.h"
+#include "src/objects/stack-frame-info.h"
+#include "src/objects/synthetic-module.h"
+#include "src/objects/templates.h"
+#include "src/objects/template-objects.h"
 #include "src/roots/roots.h"
+#include "src/wasm/wasm-objects.h"
 
 #include "torque-generated/exported-macros-assembler-tq.h"
 
-- 
2.27.0

